python/revng/scripting/projects/daemon_project.py

```diff
@@ -0,0 +1,192 @@
+            "step": step,
+            "analysis": analysis_name,
+            "index": self._get_content_commit_index(),
+        }
+
+        # NOTE: passing a single Function as `target_class` parameter
+        # causes the daemon (LLVM) to crash
+        if not target_class:
+            target_class = self.model.Functions
+
+        query = gql(
+            """
+            mutation($step: String!, $analysis: String!, $container: String!, $index: BigInt!) {
+                runAnalysis(step: $step, analysis: $analysis,
+                            containerToTargets: $container, index: $index) {
+                    __typename
+                }
+            }
+        """
+        )
+        if not container_input:
+            variables["container"] = json.dumps({})
+        else:
+            targets: dict[str, List[str]] = {}
+            for input, kinds in container_input.items():  # noqa: A001
+                targets[input] = []
+                for kind in kinds:
+                    if kind == "root":
+                        targets[input].append(f":{kind}")
+                    else:
+                        targets[input].extend(f"{tc.key()}:{kind}" for tc in target_class)
+            variables["container"] = json.dumps(targets)
+
+        self.client.execute(query, variable_values=variables)
+        self.set_model(self._get_model())
+
+    def analyses_list(self, analysis_name: str) -> None:
+        query = gql(
+            """
+            mutation($analysis: String!, $index: BigInt!) {
+                runAnalysesList(name: $analysis, index: $index) {
+                    __typename
+                }
+            }
+        """
+        )
+        variables = {"analysis": analysis_name, "index": self._get_content_commit_index()}
+
+        self.client.execute(query, variable_values=variables)
+        self.set_model(self._get_model())
+
+    def commit(self) -> None:
+        diff = make_diff(self.last_saved_model, self.model)
+        if not diff.changes:
+            return
+
+        diff_yaml = dump_diff_to_string(diff)
+
+        query = gql(
+            """
+            mutation ($options: String!, $index: BigInt!) {
+                runAnalysis(step: "initial", analysis: "apply-diff", index: $index,
+                options: $options) {
+                __typename
+                    ... on Diff {
+                        diff
+                    }
+                }
+            }
+        """
+        )
+        variables = {
+            "options": json.dumps(
+                {"apply-diff-global-name": "model.yml", "apply-diff-diff-content": diff_yaml}
+            ),
+            "index": self._get_content_commit_index(),
+        }
+
+        self.client.execute(query, variable_values=variables)
+        self.last_saved_model = self._get_model()
+
+    def _get_pipeline_description(self) -> None:
+        query = gql(
+            """
+            { pipelineDescription }
+        """
+        )
```

Comment:
Collapse on a single line `query = gql("{ pipelineDescription }")`
Ditto for `_get_model`